crypto/src/hash/prefixes.rs

```diff
@@ -84,41 +88,33 @@ pub fn make_prefix_coinbase_merkle_tree(i: usize) -> HashPrefix {
     create(base.as_bytes())
 }
 
-#[cfg(test)]
-mod tests {
-    use super::*;
-
-    #[test]
-    fn create_works_as_expected() {
-        assert_eq!(PROTOCOL_STATE.len(), 20);
-        assert_eq!(PROTOCOL_STATE, b"CodaProtoState******");
+/// Converts prefix string to field type of pasta elliptic curve
+pub fn prefix_to_field(s: &[u8]) -> Result<mina_curves::pasta::Fp, std::io::Error> {
```

Comment:
The OCaml code seems to be using base field type Fp

[ocaml code ref 1](https://github.com/MinaProtocol/mina/blob/58eea687b9c85b6b71cb447ba8a6646b818ac214/src/lib/random_oracle/random_oracle.ml#L161)
[ocaml code ref 2](https://github.com/MinaProtocol/mina/blob/58eea687b9c85b6b71cb447ba8a6646b818ac214/src/lib/random_oracle/random_oracle.ml#L91)

Reply:
I think you want `vesta::Fp` the scalar field of vesta

Reply:
I see, I will update the comment to properly reflect the fact that Fp here is the field type of vesta. Thanks!

No code change is needed here since Fp and Fq are only exported under the `pasta` namespace

Reply:
To be precise and for clarity you can import using the curve name, like
```rust
type BaseField = <mina_curves::pasta::vesta::Affine as AffineCurve>::BaseField;
```

Reply:
Fixed